File: CS/WindowsApplication3/CustomRadioGroupPainter.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using System.Drawing;
using System.Reflection;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors;
using System.ComponentModel;
using System.Windows.Forms;
using WindowsApplication3;

namespace DXSample
{
    public class CustomRadioGroupPainter : RadioGroupPainter
    {
        public CustomRadioGroupPainter() : base() { }
        protected override void DrawRadioGroupItems(ControlGraphicsInfoArgs info)
        {
            RadioGroupViewInfo vi = info.ViewInfo as RadioGroupViewInfo;
            foreach (RadioGroupItemViewInfo itemInfo in vi.ItemsInfo)
            {
                itemInfo.Cache = info.Cache;
                try
                {
                    CustomDrawEventArgs e = new CustomDrawEventArgs(info.Graphics, itemInfo, vi.RadioPainter, false);
                    RepositoryItemCustomRadioGroup item = vi.Item as RepositoryItemCustomRadioGroup;
                    item.RaiseCustomDrawItem(e);
                    if (!e.Handled)
                        vi.RadioPainter.DrawObject(itemInfo);
                    if (itemInfo.Focused)
                        ControlPaint.DrawFocusRectangle(info.Graphics, vi.GetFocusRect(itemInfo));
                }
                finally
                {
                  itemInfo.Cache = null;
                }
            }
        }
    }

}

File: VB/WindowsApplication3/CustomRadioGroupPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports System.Drawing
Imports System.Reflection
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors
Imports System.ComponentModel
Imports System.Windows.Forms
Imports WindowsApplication3

Namespace DXSample
	Public Class CustomRadioGroupPainter
		Inherits RadioGroupPainter
		Public Sub New()
			MyBase.New()
		End Sub
		Protected Overrides Sub DrawRadioGroupItems(ByVal info As ControlGraphicsInfoArgs)
			Dim vi As RadioGroupViewInfo = TryCast(info.ViewInfo, RadioGroupViewInfo)
			For Each itemInfo As RadioGroupItemViewInfo In vi.ItemsInfo
				itemInfo.Cache = info.Cache
				Try
					Dim e As New CustomDrawEventArgs(info.Cache, itemInfo, vi.RadioPainter, False)
					Dim item As RepositoryItemCustomRadioGroup = TryCast(vi.Item, RepositoryItemCustomRadioGroup)
					item.RaiseCustomDrawItem(e)
					If (Not e.Handled) Then
						vi.RadioPainter.DrawObject(itemInfo)
					End If
					If itemInfo.Focused Then
						ControlPaint.DrawFocusRectangle(info.Graphics, vi.GetFocusRect(itemInfo))
					End If
				Finally
				  itemInfo.Cache = Nothing
				End Try
			Next itemInfo
		End Sub
	End Class

End Namespace


